CHANGELOG.md

```diff
@@ -11,8 +27,41 @@
     features = ["3.10"]
     ```
 
+- Signal handlers [take the receiver by reference][gtk226-1] and support [more
+  specific receiver types][gtk226-2].
+
+- New [safe event bindings][gdk95] are [passed to signal handlers][gtk244].
+  These bindings don't have many getters and setters implemented yet but the low
+  level event structs are still reachable [via `as_ref`][examples68].
+
+- New safe [`Error`][glib86-error] and [`Value`][glib95] bindings. `ListStore` and
+  `TreeStore` [changed accordingly][gtk238], traded `set_string` for more
+  powerful [`set`][gtk241].
+
+- In an ongoing move to [reflect integer unsignedness properly][gtk246]
+  `Notebook` and partly `ListStore` and `TreeStore` switched from `i32` to
+  `u32`, `EntryBuffer` to `u16`.
+
 ### Improvements
 
+- [Object reform][gtk221] benefits include:
+
+  - Automatic generation of [the majority of code][auto-dir], expanding the
+    number of implemented classes, eliminating typos and making maintenance
+    easier.
+
+  - Unified upcasting and downcasting via the `Cast` trait, proper interface
+    support. In particular this makes `Builder` safe to use.
+
+  - The [`prelude`][prelude] module, which reexports all traits from `gtk` and
```

Comment:
This bit is slightly untrue right now but I'll open a PR to actually reexport all the dependencies' traits.